lib/api.js

```diff
@@ -48,7 +48,19 @@ function _buildContentHeaders(options) {
 }
 
 function _authenticate(options) {
-  var base64Encoded = new Buffer(options['key'] + ':').toString('base64');
+  var key = options['key'] + ':';
+  var buffer;
+
+  try {
+    buffer = Buffer.from(key);
+  } catch (e) {
+    if (e.message === 'this is not a typed array.') {
```

Comment:
Why need check this?

Reply:
`Buffer.from` suppose to officially replace `new Buffer` in v5.x, which suppose to support any parameter that is supported by `Buffer()`, however the oldest version that we still support (v4.4.6.) also implemented `Buffer.from` that which would throw `this is not a typed array.` if `key` isn't an array.

Reply:
I think this should check the existence method instead of catching an error message when method is not exist in older version of node.

Reply:
By the way, there was a backport of [Buffer API to v4.x in v4.5.0.](https://github.com/nodejs/node/pull/7562). So, the `Buffer.from()` in versions prior is not the same thing. Please checking the existence of `Buffer.allocUnsafe()` instead.
```js
// backward compatilibity for 4.x
if (typeof Buffer.allocUnsafe === "function"){
     buffer = Buffer.from(key);
} else {
     buffer = new Buffer(key);
}
```

Reply:
@aquartier Checking for existence of `Buffer.from` isn't enough, because it exist in v4 as well.

`allocUnsafe` is a good idea. Thanks